openapi.json

```
@@ -205,6 +219,20 @@
             },
             "description": "The recommendation UUID"
           },
+          {
+            "name": "k8s-units",
+            "in": "query",
+            "description": "Show values in k8s-units for recommmendation and box-plots",
+            "required": false,
+            "schema": {
+              "type": "string",
+              "enum": [
+                "true",
+                "false"
+              ],
+              "default": "true"
+            }
+          },
```

```suggestion
          {
              "name": "k8s-units",
              "in": "query",
              "description": "If true, the endpoint returns values in k8s-units formats",
              "example": false,
              "required": false,
              "allowEmptyValue": true,
              "schema": {
                    "type": "boolean",
                    "default": true
                }
          },
```
Why is it a `string` type parameter?

---

Values passed to query parameters are converted to `string` by default, from the client level.

---

Native boolean support can be done if we are accepting a json payload as such.
```
{
    "isActive": true
}
```
In case of query parameters, the values will be string when captured on the backend.

---

@kgaikwad Since the API spec is user-facing, if we denote the type as boolean then the user might get confused about the following,

1. Boolean in which language, JSON/Go/Python etc.
2. How to specify the type as boolean on the query paramm?

---

OK! In this case, will it make sense to check presence of `k8-units` query param as flag? With this there will be only two possibilities to verify - `k8-units` exist or not exist.

---

This is a possibility, but then UI will need to provide the `k8s-units` parameter on the endpoint.
Otherwise the param value would not be truthy, so we won't be able to provide k8s units as default.

---

As we want to keep k8-units as default, it makes sense to add parameter to override default behavior.  In this way, UI will not be require to include it in every request.  You can rename something as long as it conveys meaning of & intention behind this parameter to end users.

---

Yep, I have pushed the changes and updated PR description, new param is `true-units`.